Automattic/jetpack
# Update intra-monorepo deps when bumping package version

When the package version linter detects you need to bump a package's version, it recommends that you run something like `tools/project-version.sh -u 0.3.1-alpha js-packages/components`.

This does not update the versions of any other packages using that package. Yet, such a script exists! It's `tools/check-intra-monorepo-deps.sh -u`!

We don't always want to update other deps, so this patch adds an optional extra `-f` argument to `tools/check-intra-monorepo-deps.sh` and only recommends it when you are manually bumping the project version from a linter error. This preserves existing behaviour in all the places that we call this from a script.

#### Changes proposed in this Pull Request:
* Add optional parameter to tools/project-version.sh which will automatically check and fix intra-monorepo dependencies.

#### Does this pull request change what data or activity we track or use?
No

#### Testing instructions:

Check out a new branch from this one:

```
git checkout -b test/update-intra-monorepo-deps
```

Bump a JS package:

```
tools/project-version.sh -f -u 0.3.1-alpha js-packages/components
```

Check that all the references to `@automattic/jetpack-components` have been updated, as well as the package itself.

Bump a PHP package:

```
tools/project-version.sh -f -u 1.17.0 packages/tracking
```

Check that all the references to `automattic/jetpack-tracking` have been updated, as well as the package itself.
